pkg/generate/code/compare_helpers.go

```diff
@@ -0,0 +1,460 @@
+//  		if *va.DetailedMetricsEnabled != *vb.DetailedMetricsEnabled {
+//  			return false
+//  		}
+//  	}
+//  	if ackcompare.HasNilDifference(va.LoggingLevel, vb.LoggingLevel) {
+//  		return false
+//  	} else if va.LoggingLevel != nil && vb.LoggingLevel != nil {
+//  		if *va.LoggingLevel != *vb.LoggingLevel {
+//  			return false
+//  		}
+//  	}
+//  	if ackcompare.HasNilDifference(va.ThrottlingBurstLimit, vb.ThrottlingBurstLimit) {
+//  		return false
+//  	} else if va.ThrottlingBurstLimit != nil && vb.ThrottlingBurstLimit != nil {
+//  		if *va.ThrottlingBurstLimit != *vb.ThrottlingBurstLimit {
+//  			return false
+//  		}
+//  	}
+//  	if ackcompare.HasNilDifference(va.ThrottlingRateLimit, vb.ThrottlingRateLimit) {
+//  		return false
+//  	} else if va.ThrottlingRateLimit != nil && vb.ThrottlingRateLimit != nil {
+//  		if *va.ThrottlingRateLimit != *vb.ThrottlingRateLimit {
+//  			return false
+//  		}
+//  	}
+//  }
+func equalMap(
+	// struct describing the SDK type of the field being compared
+	shape *awssdkmodel.Shape,
+	// String representing the name of the variable that represents the object
+	// under comparison.
+	firstVarName string,
+	// String representing the name of the variable that represents the second
+	// object under comparison.
+	secondVarName string,
+	// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	out := ""
+	indent := strings.Repeat("\t", indentLevel)
+	memberValueShape := shape.ValueRef.Shape
+
+	// 	if len(a) != len(b) {
+	out += fmt.Sprintf(
+		"%sif len(%s) != len(%s) {\n",
+		indent,
+		firstVarName,
+		secondVarName,
+	)
+	// return false
+	out += fmt.Sprintf(
+		"%s\treturn false\n", indent,
+	)
+	// }
+	out += fmt.Sprintf("%s}\n", indent)
+
+	// for ka := range a {
+	out += fmt.Sprintf(
+		"%sfor ka := range %s {\n",
+		indent,
+		firstVarName,
+	)
+	// _, ok := b[ka]
+	out += fmt.Sprintf(
+		"%s\t_, ok := %s[ka]\n",
+		indent,
+		secondVarName,
+	)
+	// if !ok {
+	out += fmt.Sprintf("%s\tif !ok {\n", indent)
+	// return false
+	out += fmt.Sprintf(
+		"%s\t\treturn false\n", indent,
+	)
+	// }
+	out += fmt.Sprintf("%s\t}\n", indent)
+	// closing the for loop
+	out += fmt.Sprintf("%s}\n", indent)
+
+	firstVarAdaptedName := firstVarName + "X"
+	secondVarAdaptedName := secondVarName + "Y"
+	// for ka, kv := range b {
+	out += fmt.Sprintf("%sfor ka, %s := range a {\n", indent, firstVarAdaptedName)
+	// vb := b[ka]
+	out += fmt.Sprintf("%s\t%s := b[ka]\n", indent, secondVarAdaptedName)
+
+	switch memberValueShape.Type {
+	case "string":
+		// if !ackcompare.MapStringStringPEqual(a.ko.Spec.Tags, b.ko.Spec.Tags) {
+		out += fmt.Sprintf(
+			"%sif !ackcompare.MapStringStringPEqual(%s, %s) {\n",
+			indent, firstVarAdaptedName, secondVarAdaptedName,
+		)
+		out += fmt.Sprintf(
+			"%s\n", indent,
+		)
+	case "structure":
+		out += equalStruct(
+			memberValueShape,
+			firstVarAdaptedName,
+			secondVarAdaptedName,
+			indentLevel+1,
+		)
+	case "list", "blop":
```

Comment:
what a 'blop' shape?

Reply:
Yikes! I meant "blob"... "blop" is a dungeon in a game i used to play long time ago ^^'